### PR TITLE
SRVCOM-1359: Use Serverless's showcase app to showcase Serving & Eventing

### PR DESCRIPTION
**Pull request by cardil**
Version(s):
OCP 4.10+

In scope of Serverless 1.30

Issue:
https://issues.redhat.com/browse/SRVCOM-1359

Link to docs preview:
https://60772--docspreview.netlify.app/

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by cardil** [comment]
/hold

To be in scope of Serverless 1.30

File: modules/apiserversource-kn.adoc
Comment (mgencur) [inline modules/apiserversource-kn.adoc:92]: We didn't mention "event-display" service anywhere before. We suggested exactly this: `kn service create <service_name> --image quay.io/openshift-knative/showcase` . So, maybe we could suggest a concrete service name at the beginning and at this point use the concrete name as well.

**Review by mgencur — COMMENTED** [review]
Hey @cardil , that looks pretty good! 
I am just confused if we want to use "event-display" name or "showcase" name throughout the docs. In some documents you renamed it to "showcase", in some other docs it stays as "event-display". And in some docs the service is called "showcase" and then we reference it as "event-display". We just need to unify it everywhere, IMO.

File: modules/apiserversource-kn.adoc
Comment (mgencur) [reply modules/apiserversource-kn.adoc:92]: I.e. let's use this at the beginning: `kn service create event-display --image quay.io/openshift-knative/showcase` and then we can reference it as "event-display" later.

**Comment by mgencur** [comment]
> WDYT?

Yeah. OK. That sounds fine. I have deleted some of my comment and left just those that are still valid.

File: modules/apiserversource-yaml.adoc
Comment (cardil) [reply modules/apiserversource-yaml.adoc:102]: How about we name this something like: "event-origin"?

File: modules/apiserversource-yaml.adoc
Comment (cardil) [reply modules/apiserversource-yaml.adoc:154]: In fact, in this case (YAML version of this doc), we are already using "event-display" name: https://github.com/cardil/openshift-docs/blob/2365a29d87fcc3f0774151d645f4d37464cb955c/modules/apiserversource-yaml.adoc?plain=1#L56-L58

File: modules/apiserversource-kn.adoc
Comment (mgencur) [reply modules/apiserversource-kn.adoc]: > I renamed it to: event-origin

Sounds good.

File: modules/apiserversource-yaml.adoc
Comment (mgencur) [reply modules/apiserversource-yaml.adoc:154]: OK. I overlooked it, thanks.

File: modules/apiserversource-yaml.adoc
Comment (mgencur) [reply modules/apiserversource-yaml.adoc]: Yeah. OK.

**Comment by mgencur** [comment]
LGTM from QE

**Comment by mgencur** [comment]
There's now https://github.com/openshift-knative/docs/pull/95/files in flight which also uses the old image (not showcase). Before merging the current PR we should check if the other PR is already merged and probably fix those places as well.

File: modules/apiserversource-kn.adoc
Comment (nalhadef) [inline modules/apiserversource-kn.adoc:97]: This (and a couple lines after this) seems to cover the same ground as the lines in the late 80s above. Can they be consolidated?

File: modules/apiserversource-kn.adoc
Comment (nalhadef) [inline modules/apiserversource-kn.adoc:86]: Delete "you can."

File: modules/creating-serverless-apps-kn.adoc
Comment (nalhadef) [inline modules/creating-serverless-apps-kn.adoc:45]: Should route and revision have initial caps?

File: modules/serverless-domain-mapping-odc-developer.adoc
Comment (nalhadef) [inline modules/serverless-domain-mapping-odc-developer.adoc:26]: Delete "on," leaving "Right clickthe service you want...."

File: modules/verifying-serverless-app-deployment.adoc
Comment (nalhadef) [inline modules/verifying-serverless-app-deployment.adoc:65]: I'd move "likely" so it's after "can."

**Comment by nalhadef** [comment]
Looks good @cardil. I added a handful of suggestions.

File: modules/creating-serverless-apps-kn.adoc
Comment (cardil) [reply modules/creating-serverless-apps-kn.adoc:45]: This is an output of `kn` command, so we can't change it here.

We can propose to change the `kn` command itself, in [its repo](https://github.com/knative/client). I think the Knative contributors wrote those with caps, as being a Knative's proper names.

File: modules/apiserversource-kn.adoc
Comment (cardil) [reply modules/apiserversource-kn.adoc:97]: I don't get your proposal, would you kindly specify it more clearly?

My understanding:

Paragraph at 86 describes there are 2 ways of verification: in-browser, or logs

Lines 88-95 shows the first way of using the web browser to do the verification.

Lines 97-135 shows the second method - the logs.

File: modules/verifying-serverless-app-deployment.adoc
Comment (cardil) [reply modules/verifying-serverless-app-deployment.adoc]: Done.

File: modules/serverless-domain-mapping-odc-developer.adoc
Comment (cardil) [reply modules/serverless-domain-mapping-odc-developer.adoc]: Done.

File: modules/apiserversource-kn.adoc
Comment (nalhadef) [reply modules/apiserversource-kn.adoc:97]: The words for verifying via the even logs in lines 86 and 97 are almost identicle. nothing new is added to give value to the reader. My suggestion is to move the command from Line 86 to Line 97.

File: modules/creating-serverless-apps-kn.adoc
Comment (nalhadef) [reply modules/creating-serverless-apps-kn.adoc:45]: No worries. Thanks

**Comment by Preeticp** [comment]
Hey @nalhadef could you PTAL and provide a LGTM if you think things are good, this is due for tomorrow and we would like to push it to the merge review stage as soon as possible. If you are not on the roster, please let us know if this needs to be passed on to someone else. Thank you!

File: modules/apiserversource-kn.adoc
Comment (cardil) [reply modules/apiserversource-kn.adoc:97]: The procedure from 88-95 shows hot to get the URL and how that web page should look like (image attached).

The second procedure 97-135 shows different command to display logs in the terminal. It's entirely different.

Comment (cardil) [reply modules/apiserversource-kn.adoc:97]: I don't see a clear way on how I could fold those procedures into one, while keeping the current meaning.

Please, maybe you could suggest a wording you have in mind...

File: modules/apiserversource-yaml.adoc
Comment (mburke5678) [inline modules/apiserversource-yaml.adoc:156]: Need a `+` to keep the image in line with the text.
```suggestion
+
```
The missing `+` also affects the numbering. See [Verification in the preview](https://60772--docspreview.netlify.app/openshift-serverless/latest/eventing/event-sources/serverless-apiserversource#apiserversource-yaml_context).

File: modules/apiserversource-kn.adoc
Comment (mburke5678) [inline modules/apiserversource-kn.adoc:96]: Need a `+` to keep the image in line with the text.
```suggestion
+
```
The missing `+` also affects the numbering. See [Verification in the preview](https://60772--docspreview.netlify.app/openshift-serverless/latest/eventing/event-sources/serverless-apiserversource#apiserversource-kn_serverless-apiserversource)

File: modules/apiserversource-kn.adoc
Comment (mburke5678) [reply modules/apiserversource-kn.adoc:97]: @cardil At the least, you could change the numbered list to unordered, as the two methods are not performed sequentially -- I should think most users would do one of the other and not both.

File: modules/apiserversource-yaml.adoc
Comment (mburke5678) [inline modules/apiserversource-yaml.adoc:150]: Please change the numbered list to unordered, as the two methods are not performed sequentially -- I should think most users would do one of the other and not both.

Comment (cardil) [reply modules/apiserversource-yaml.adoc:150]: How about changing this like here: https://github.com/openshift/openshift-docs/compare/103534fc90ce678a24d7a257f8732e0ffb2c2da1..37cfb46f19e87d2659e953dc1bc3e1781ffc3e4b - to have two verification sections?

Comment (mburke5678) [reply modules/apiserversource-yaml.adoc:150]: I should think something this would be fine:

```
.Verification

To verify that the Kubernetes events were sent to Knative, you can look at the event-display logs or use web browser to see the events.

. To view the events in a web browser, open the link returned by the following command:
+
[source,terminal]
----
$ oc get ksvc event-display -o jsonpath='{.status.url}'
----
+
. Example browser page
+
image::serverless-apiserversource-eventdisplay-gui.png[Example visualization of ApiServerSource event]

. To see the logs in the terminal, view the event-display logs for the pods by entering the following command:
+
[source,terminal]
----
$ oc logs $(oc get pod -o name | grep event-display) -c user-container
----
+
.Example output
. Example output
[source,terminal]
```

Comment (cardil) [reply modules/apiserversource-yaml.adoc:150]: That would end up with a numbered list, doesn't it?

> 1. To view..
> 2. Example browser page
> 3. To see the logs..
> 4. Example output

I thought you were saying that it's confusing, and I agree.

Comment (mburke5678) [reply modules/apiserversource-yaml.adoc:150]: @cardil Yore'right. I'm sorry.  Should be `*` not `.`.

.Verification

To verify that the Kubernetes events were sent to Knative, you can look at the event-display logs or use web browser to see the events.

* To view the events in a web browser, open the link returned by the following command:
+
[source,terminal]
----
$ oc get ksvc event-display -o jsonpath='{.status.url}'
----
+
.Example browser page
+
image::serverless-apiserversource-eventdisplay-gui.png[Example visualization of ApiServerSource event]

* To see the logs in the terminal, view the event-display logs for the pods by entering the following command:
+
[source,terminal]
----
$ oc logs $(oc get pod -o name | grep event-display) -c user-container
----
+
.Example output
[source,terminal]

Comment (cardil) [reply modules/apiserversource-yaml.adoc]: Done it. There was the same issue in the other file. Fixed that in https://github.com/openshift/openshift-docs/compare/bd304c54ba47f08ec34a1919abe3b78a0c333189..8f4a5f2d3455549ca9909fdbbd2c985f55bc9612

**Comment by Preeticp** [comment]
@mburke5678 can you PTAL again and merge if the PR looks good to you?

**Review by mburke5678 — APPROVED** [review]

**Comment by mburke5678** [comment]
@cardil I am not clear on which Serveless branches to cherry pick this one to. Can you please advise?

**Comment by cardil** [comment]
Let's push this to the Serverless 1.30 release for now.

Maybe we can back port this to 1.28 and 1.29 (the original intent) - to be confirmed...

**Comment by Preeticp** [comment]
Hey aren't the CP already done to 1.28; 1.29 and 1.30?
https://github.com/openshift/openshift-docs/pull/64252,
https://github.com/openshift/openshift-docs/pull/64253
https://github.com/openshift/openshift-docs/pull/64254
I think they just need to be merged

**Pull request merged**
